src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -370,6 +363,51 @@ class Manager : public DoutPrefixProvider {
     return ret;
   }
 
+  int create_push_endpoint(const std::string &queue_name,
+                           const std::vector<cls_queue_entry> &entries,
+                           spawn::yield_context yield,
+                           RGWPubSubEndpoint::Ptr *push_endpoint) {
```

Comment:
why passing a pointer to a unique_ptr?
you can just pass the bare pointer from the unique_ptr

Reply:
or even better. since the caller ignore the return value from this function, you can return the unique_ptr.
you can return a unique_ptr initialized with nullptr in case of error

Reply:
the caller does honor the return value, we do not process the entries if we cannot create the push_endpoint.
and i am passing the unique_ptr by ref and its created inside the function.
i do not want to verify based on whether the function returned nullptr

Reply:
ok. so please pass by ref, not a pointer. IMO, pointer to a unique_ptr is too confusing.

Reply:
remove the code to pass unique_ptr